### PR TITLE
[Add] Add dark mode logic

### PR DESCRIPTION
**Pull request by JaeHeon-Kim**
## Description
riverpod과 shared_preference를 활용해 dark mode 초기 구조 작성
 
## Changes Made
- lib/main.dart: riverpod, shared preference, Firebase.initializeApp 등록
- lib/provider/shared_preference_provider.dart: SharedPreferenceProvider 추가
- lib/provider/theme_provider.dart: ThemeProvider 추가
- lib/view/write_screen.dart/write_page.dart: WritePage 추가
- lib/widgets/constants/app_theme.dart: AppTheme 추가

## Screenshots

https://user-images.githubusercontent.com/76954805/223158262-ac5af7de-efc6-4627-8cf7-07ee7295d418.mp4

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,120 +1,36 @@
...
 void main() async {
   await Firebase.initializeApp(
     options: DefaultFirebaseOptions.currentPlatform,
   );
-  runApp(const MyApp());
-}
 
-class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+  WidgetsFlutterBinding.ensureInitialized();
```
Comment (Lukaid-dev) [inline lib/main.dart]: WidgetsFlutterBinding.ensureInitialized(); 얘는 main 의 최상단에 있어야 합니다

Comment (JaeHeon-Kim) [reply lib/main.dart]: 해결했습니다!

**Review by Lukaid-dev — APPROVED** [review]

**Review by illinpilli — APPROVED** [review]
Love ya!

File: lib/main.dart (hunk 1 above)
Comment (ChoiSeungHyun9662) [reply lib/main.dart]: 첫 코멘트

**Pull request merged**